### ISSUE TITLE
BPSec decoding mishandles larger parameter and result sizes

### ISSUE THREAD

**Issue by BrianSipos** [nasa/HDTN#54]
For some reason there is a case where the size of a decoded chunk of BTSD is cast down to an 8-bit value, when the size could actually be arbitrarily large. This causes later parts of the block to fail to decode and/or indicates that decoding failed even if it did not.

A patch to remove the cast and fix this problem is below.

```diff
--- a/common/bpcodec/src/codec/Bpv7BpSecExtensionBlocks.cpp
+++ b/common/bpcodec/src/codec/Bpv7BpSecExtensionBlocks.cpp
@@ -597,7 +597,7 @@ bool Bpv7AbstractSecurityBlock::DeserializeIdValuePairBpv7(uint8_t * serializati
         }
     }
 
-    numBytesTakenToDecode = static_cast<uint8_t>(serialization - serializationBase);
+    numBytesTakenToDecode = (serialization - serializationBase);
     return true;
 }
 bool Bpv7AbstractSecurityBlock::IsEqual(const id_value_pairs_vec_t & pVec1, const id_value_pairs_vec_t & pVec2) {
```

**Comment by rdudukov** [comment]
Hi Brian, we will look into this and get back to you. Thanks!

**Comment by briantomko** [comment]
This issue has been resolved by the merge of commit 88f579e0 into development.
